src/controller/backend/Endpoints/UserManagementEndpoints.cs

```diff
@@ -0,0 +1,104 @@
+﻿using ApiSchema.Settings;
+using ApiSchema.Usermanagement;
+using Data.System;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Backend.Endpoints;
+
+public static class UserManagementEndpoints
+{
+    public static void MapUserManagementEndpoints(this IEndpointRouteBuilder endpoints)
+    {
+        var group = endpoints.MapGroup("/api/usermanagement");
+
+        group.MapGet("/username", async (
+            HttpContext context,
+            UserManager<SystemUser> userManager) =>
+        {
+            var user = await userManager.GetUserAsync(context.User);
+
+            if (user is null)
+                return Results.Unauthorized();
+
+            return Results.Ok(user.UserName!);
+
+        }).RequireAuthorization("token");
+
+        group.MapPost("/username", async (
+            HttpContext context,
+            UserManager<SystemUser> userManager,
+            [FromBody] UsernameModel model) =>
+        {
+            var user = await userManager.GetUserAsync(context.User);
+            if (user is null)
+                return Results.Unauthorized();
+
+            if(user.UserName == "admin")
+                return Results.BadRequest("Username for the \"admin\" user cannot be changed");
```

Comment:
This is fine for now, but maybe we should make an "Internal" flag for all internal user accounts that can't be modified, like the admin or maybe a visitor or blank access testing account

Reply:
I will mark it as todo